pkg/apis/core/v1beta1/types_utils.go

```diff
@@ -69,4 +69,7 @@ const (
 	// OutdatedStatusError is a constant for a reason in a condition that indicates
 	// that the observed generation in a status is outdated.
 	OutdatedStatusError = "OutdatedStatus"
+	// ManagedResourceStuckInProgressingError is a constant for a reason in a condition that indicates
+	// managed resource progressing condition is stuck in the true state for more than the threshold time.
+	ManagedResourceStuckInProgressingError = "ManagedResourceStuckInProgressing"
```

Comment:
```suggestion
	ManagedResourceStuckInProgressingError = "ProgressingRolloutStuck"
```

Maybe this or `ProgressingStuck` is clearer for a "reason" - the message can then contain all the details why.